sources/sql_database/helpers.py

```diff
@@ -254,6 +254,36 @@ def _unwrap(table: TDataItem) -> TDataItem:
     return _unwrap
 
 
+def validate_reflection_level(
+    reflection_level: Optional[ReflectionLevel], backend: TableBackend
+) -> ReflectionLevel:
+    if backend == "pyarrow":
+        if reflection_level is None:
+            return "full"
+        if reflection_level == "minimal":
+            raise ValueError(
+                (
+                    "pyarrow backend requires reflection_level to be 'full' or 'full_with_precision' to infer schema. "
```

Comment:
why? schema is optional in https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.from_pydict

what you should do (see my other comments):
1. create arrow table for known data types
2. create arriow table for unknown data types where pyarrow infers the schema from the data
3. create unified table (preserving original column order)

anyway: you have unknown data types, in minimal all data types are uknown ;>

Reply:
That's a good idea!  
We do have some problem with objects that pyarrow doesn't support though. E.g. UUID and some dataclass objects that sqlalchemy returns for `Range` types are rejected, they need to be coerced to str/dict first.   
But I think there's not that many such types (built into sqlalchemy at least) so we could run just those columns through some conversion.